uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/manager-leave/components/manager-leave-calendar_test.js

```diff
@@ -84,6 +85,32 @@
           return $q.resolve(data);
         });
 
+        /**
+         * This is unfortunately a hack to make sure that
+         * all mocked contacts have a mocked calendar
+         *
+         * It takes the mocked WorkPattern.getcalendar response and expands it
+         * by making copies of the first mocked calendar and assigning it to
+         * each mocked contacts.
+         *
+         * The ideal solution would be to decouple the mocked data from the
+         * `api.contact.mock` (common/mocks/services/api/contact-mock) and put it
+         * in a separate file, so that mocks/data/work-pattern-data can inject it
+         * and dynamically generate the data by looping through all mocked contacts
+         */
+        spyOn(WorkPatternAPI, 'getCalendar').and.callFake(function () {
+          var mockedGetCalendarResponse = _.clone(workPatternMocked.getCalendar);
+          var singleMockedCalendarTemplate = mockedGetCalendarResponse.values[0];
+
+          return $q.resolve(_.assign(mockedGetCalendarResponse, {
+            values: ContactAPIMock.mockedContacts().list.map(function (contact) {
+              var clonedCalendar = _.clone(singleMockedCalendarTemplate);
```

Comment:
`return _.assign(_.clone(singleMockedCalendarTemplate), { contact_id: contact.id });` is better in this case